pkg/kubelet/volume/volume_manager.go

```diff
@@ -209,16 +213,42 @@ func (vm *volumeManager) GetVolumesForPodAndAppendSupplementalGroups(
 }
 
 func (vm *volumeManager) GetVolumesInUse() []api.UniqueVolumeName {
-	attachedVolumes := vm.actualStateOfWorld.GetAttachedVolumes()
-	volumesInUse :=
-		make([]api.UniqueVolumeName, 0 /* len */, len(attachedVolumes) /* cap */)
-	for _, attachedVolume := range attachedVolumes {
-		if attachedVolume.PluginIsAttachable {
-			volumesInUse = append(volumesInUse, attachedVolume.VolumeName)
+	// Report volumes in desired state of world and actual state of world so
+	// that volumes are marked in use as soon as the decision is made that the
+	// volume *should* be attached to this node until it is safely unmounted.
+	desiredVolumes := vm.desiredStateOfWorld.GetVolumesToMount()
```

Comment:
Correct. We have an issue https://github.com/kubernetes/kubernetes/issues/27653 open to handle restarts more gracefully.